pkg/controller/datadogagentdeployment/poddisruptionbudget.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package datadogagentdeployment
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	datadoghqv1alpha1 "github.com/DataDog/datadog-operator/pkg/apis/datadoghq/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1beta1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	pdbMinAvailableInstances = 1
+)
+
+type (
+	pdbBuilder func(dad *datadoghqv1alpha1.DatadogAgentDeployment) *policyv1.PodDisruptionBudget
+)
+
+func (r *ReconcileDatadogAgentDeployment) manageClusterAgentPDB(logger logr.Logger, dad *datadoghqv1alpha1.DatadogAgentDeployment, newStatus *datadoghqv1alpha1.DatadogAgentDeploymentStatus) (reconcile.Result, error) {
+	cleanUpCondition := dad.Spec.ClusterAgent == nil
+	return r.managePDB(logger, dad, newStatus, getClusterAgentPDBName(dad), buildClusterAgentPDB, cleanUpCondition)
+}
+
+func (r *ReconcileDatadogAgentDeployment) manageClusterChecksRunnerPDB(logger logr.Logger, dad *datadoghqv1alpha1.DatadogAgentDeployment, newStatus *datadoghqv1alpha1.DatadogAgentDeploymentStatus) (reconcile.Result, error) {
+	cleanUpCondition := !needClusterChecksRunner(dad)
+	return r.managePDB(logger, dad, newStatus, getClusterChecksRunnerPDBName(dad), buildClusterChecksRunnerPDB, cleanUpCondition)
+}
+
+func (r *ReconcileDatadogAgentDeployment) managePDB(logger logr.Logger, dad *datadoghqv1alpha1.DatadogAgentDeployment, newStatus *datadoghqv1alpha1.DatadogAgentDeploymentStatus, pdbName string, builder pdbBuilder, cleanUp bool) (reconcile.Result, error) {
+	if cleanUp {
+		result, err := r.cleanupPDB(logger, dad, newStatus, pdbName)
+		return result, err
```

Comment:
maybe we can `return r.cleanupPDB(logger, dad, newStatus, pdbName)` directly

Reply:
Done